src/js/common/utils/accessible-menus.js

```diff
@@ -0,0 +1,349 @@
+    return;
+  }
+
+  const { menuButton, menu } = struct;
+
+  // Close the menu
+  menuButton.removeAttribute('aria-expanded');
+  menu.setAttribute('hidden', 'hidden');
+}
+
+
+/**
+ * Opens a menu.
+ *
+ * @param {HTMLLIElement} menuLi  The <li> containing the menubutton and menu
+ * @param {bool} openSubMenu      Whether or not the first submenu should be opened. E.g. if we're opening the
+ *                                 second-level menu in a wide screen, we want the third-level opened as well
+ * @param {bool} stealFocus       Whether or not to focus on the first item in the opened menu
+ */
+function openMenu(menuLi, openSubMenu = false, stealFocus = true) {
+  // If we're not dealing with a menu structure, abort
+  const struct = getMenuStructure(menuLi);
+  if (!struct) {
+    return;
+  }
+
+  const { menuButton, menu } = struct;
+
+  // First, close all sibling menus
+  const openMenus = menuLi.parentElement.querySelectorAll('[aria-expanded=true]');
+  openMenus.forEach(m => closeMenu(m.parentElement));
+
+  // Open the menu
+  menuButton.setAttribute('aria-expanded', true);
+  menu.removeAttribute('hidden');
+
+  if (stealFocus) {
+    const menuRole = (menu.getAttribute('role') || '').toLowerCase();
+    const firstMenuItem = menuRole === 'menu' ?
+      firstVisibleChild(menu) :
+      firstVisibleChild(menu.querySelector('[role="menu"]'));
+    firstMenuItem.firstElementChild.focus();
+  }
+
+  // If we're wide-screen (for example), open the first submenu
+  if (openSubMenu) {
+    openMenu(menu.firstElementChild, false, false);
+  }
+}
+
+
+/**
+ * Creates a closure which closes all menus in the provided element.
+ *
+ * @param {HTMLElement}
+ * @return {function}   Call this to close all menus
+ */
+function closeAll(menuElement) {
+  // Get all the parent <li>s of open menus
+  const allMenus = Array.from(menuElement.querySelectorAll('[aria-expanded=true]'))
+    .map(e => findNearestLi(e));
+
+  // Add the top-level 
+  if (menuElement.getAttribute('aria-expanded') === true) {
+    allMenus.push(findNearestLi(menuElement));
+  }
+
+  allMenus.forEach(e => closeMenu(e));
+}
+
+
+/**
+ * Attaches event listeners to a menu or menu bar to make it keyboard navigable.
+ * 
+ * If the mobile buttons are provided, this will also handle the opening and closing
+ *  of the menu on small screens when there's just a "Menu" button.
+ *
+ * @param {HTMLElement} menuElement        A menubar or menu
+ * @param {HTMLElement} mobileOpenButton   Optional - The "Menu" button on mobile
+ * @param {HTMLElement} mobileCloseButton  Optional - The "Close" button on mobile
+ */
+export default function addMenuListeners(menuElement) {
+  const menuRole = menuElement.getAttribute('role');
+  if (!['menubar', 'menu'].includes(menuRole)) {
+    // If we don't have a menubar or menu, don't continue
+    return;
+  }
+
+  // For all the sub menus, add listeners for:
+  //  Up, down, left, right, enter (if necessary), space (if necessary)
```

Comment:
😬

Reply:
Ahh, right. Make like a Cyberman and DELETE!